### PR TITLE
Fboemer/windows ci

### PR DESCRIPTION
**Pull request by fboemer**
Initial pass at adding Windows CI. Some issues with the debug build, so skipping for now. For the most part, the Windows runners seem to have AVX512, though I don't believe this is guaranteed.

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -39,8 +39,9 @@ elseif (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
     endif()
     # Disable inline, due to incorect optimization in ExtractValues, causing failing tests in Windows AVX512 in Release mode with HEXL_DEBUG=OFF
     target_compile_options(unit-test PRIVATE /Wall /W4 /Zc:preprocessor /Ob0
-        /wd4127 # warning C4127: conditional expression is constant; C++11 doesn't support constexpr
+        /wd4127 # warning C4127: conditional expression is constant; C++11 doesn't support if constexpr
         /wd4389 # warning C4389: signed/unsigned mismatch from gtest
+        /wd5105 # warning C5105: macro expansion prooducing 'defined' has undefined behavior
```
Comment (jlhcrawford) [inline test/CMakeLists.txt:44]:
```suggestion
        /wd5105 # warning C5105: macro expansion producing 'defined' has undefined behavior
```

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -39,8 +39,9 @@ elseif (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
     endif()
     # Disable inline, due to incorect optimization in ExtractValues, causing failing tests in Windows AVX512 in Release mode with HEXL_DEBUG=OFF
```
Comment (jlhcrawford) [inline test/CMakeLists.txt:40]:
```suggestion
    # Disable inline, due to incorrect optimization in ExtractValues, causing failing tests in Windows AVX512 in Release mode with HEXL_DEBUG=OFF
```

File: hexl/CMakeLists.txt [hunk 3]
```diff
@@ -77,7 +77,9 @@ elseif (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
     # Inling causes some tests to fail on MSVC with AVX512 in Release mode, HEXL_DEBUG=OFF,
     # so we disable it here
     target_compile_options(hexl PRIVATE /Wall /W4 /Zc:preprocessor /Ob0
-        /wd4127 # warning C4127: conditional expression is constant; C++11 doesn't support constexpr
+        /wd4127 # warning C4127: conditional expression is constant; C++11 doesn't support if constexpr
+        /wd5105 # warning C5105: macro expansion prooducing 'defined' has undefined behavior
```
Comment (jlhcrawford) [inline hexl/CMakeLists.txt:81]:
```suggestion
        /wd5105 # warning C5105: macro expansion producing 'defined' has undefined behavior
```

File: hexl/CMakeLists.txt [hunk 4]
```diff
@@ -77,7 +77,9 @@ elseif (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
     # Inling causes some tests to fail on MSVC with AVX512 in Release mode, HEXL_DEBUG=OFF,
```
Comment (jlhcrawford) [inline hexl/CMakeLists.txt:77]:
```suggestion
    # Inlining causes some tests to fail on MSVC with AVX512 in Release mode, HEXL_DEBUG=OFF,
```

File: benchmark/CMakeLists.txt [hunk 5]
```diff
@@ -29,5 +29,6 @@ if (CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
 elseif (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
     target_compile_options(bench_hexl PRIVATE /Wall /W4 /Zc:preprocessor
         /wd4127 # warning C4127: conditional expression is constant; C++11 doesn't support constexpr
+        /wd5105 # warning C5105: macro expansion prooducing 'defined' has undefined behavior
```
Comment (jlhcrawford) [inline benchmark/CMakeLists.txt:32]:
```suggestion
        /wd5105 # warning C5105: macro expansion producing 'defined' has undefined behavior
```

File: benchmark/CMakeLists.txt [hunk 6]
```diff
@@ -29,5 +29,6 @@ if (CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
 elseif (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
     target_compile_options(bench_hexl PRIVATE /Wall /W4 /Zc:preprocessor
         /wd4127 # warning C4127: conditional expression is constant; C++11 doesn't support constexpr
```
Comment (jlhcrawford) [inline benchmark/CMakeLists.txt:31]:
```suggestion
        /wd4127 # warning C4127: conditional expression is constant; C++11 doesn't support if constexpr
```

File: README.md [hunk 7]
```diff
@@ -134,14 +134,19 @@ Then, run
 ```bash
 cmake --build build --target unittest
 ```
-The unit-test executable itself is located at `build/test/unit-test`
+The unit-test executable itself is located at `build/test/unit-test` on Linux and Mac, and at `build\test\Release\unit-test.exe` or `build\test\Debug\unit-test.exe`.
```
Comment (jlhcrawford) [inline README.md:137]:
```suggestion
The unit-test executable itself is located at `build/test/unit-test` on Linux and Mac, and at `build\test\Release\unit-test.exe` or `build\test\Debug\unit-test.exe` on Windows.
```

File: .github/workflows/github-ci.yml [hunk 8]
```diff
@@ -72,7 +72,58 @@ jobs:
         with:
           extra_args: --all-files
 
-  default-build:
+  windows-release-build:
+    name: Windows release
```
Comment (jlhcrawford) [inline .github/workflows/github-ci.yml:76]:
```suggestion
    name: Windows release
    needs: [format]
```
Is this job supposed to require the code be formatted beforehand like the linux builds?

File: .github/workflows/github-ci.yml [hunk 9]
```diff
@@ -72,7 +72,58 @@ jobs:
...
+
+          # Build and run examples
+          cd example/cmake
+          cmake -S . -B build ${{ env.DEFAULT_EXAMPLE_FLAGS }}
+          cmake --build build --config Release
+          build/Release/example.exe
+
+  # windows-debug-build:
+  #   name: Windows debug
+  #   runs-on: windows-latest
```
Comment (jlhcrawford) [inline .github/workflows/github-ci.yml:102]: Same as above `needs: [format]` ?

**Review by jlhcrawford — CHANGES_REQUESTED** [review]
Some typos and a question about the Windows CI.

File: hexl/CMakeLists.txt (hunk 3 above)
Comment (jlhcrawford) [reply hexl/CMakeLists.txt:81]: This typo is still here.

**Review by jlhcrawford — CHANGES_REQUESTED** [review]
You missed a typo, otherwise it looks good.

**Comment by fboemer** [comment]
> You missed a typo, otherwise it looks good.

My mistake, thanks

**Review by jlhcrawford — APPROVED** [review]
LGTM

**Review by ajagann — APPROVED** [review]
LGTM :)

**Pull request merged**